RFC0032.md

```diff
@@ -0,0 +1,112 @@
+# RFC0032
+Extends:  
+* [RFC0010: Extension of Tool Interaction Support](https://github.com/pmix/RFCs/blob/master/RFC0010.md)
+  * Adds attributes to broaden logging support
+  * Defines a "gateway" server type
+
+## Title
+Extension of PMIx Logging Support
+
+## Abstract
+The PMIx\_Log API supports requests for logging information to a variety of channels. Initial support was only for stdout/stderr - this RFC extends that support to additional channels such as syslog and smtp.
+
+## Labels
+[EXTENSION][ATTRIBUTES]
+
+## Action
+[SUBMITTED]
+
+## Copyright Notice
+Copyright (c) 2018 Intel, Inc. All rights reserved.
+
+This document is subject to all provisions relating to code contributions to the PMIx community as defined in the community's [LICENSE](https://github.com/pmix/RFCs/tree/master/LICENSE) file. Code Components extracted from this document must include the License text as described in that file.
+
+## Description
+RFC0010 added a new PMIx\_Log\_nb API for requesting logging of provided data in some global data store or to standard output locations (e.g., stderr or syslog), subject to available services from the host environment. A limited number of initial keys were provided by which the logging request could direct the data to specific channels on an as-available basis, and the host RM interface was extended through the addition of a "log" interface by which client requests that are not directly supported by the PMIx library can be passed to the host RM for handling - e.g., when the logging data is directed to the stderr or stdout channels.
+
+The illustration below highlights some of the capabilities provided by the extended logging support. Note that the illustration is not intended to be comprehensive in its coverage as the number of possible use-cases is too large to capture in a single drawing.
+
+<img src="images/rfc0032/logging.png"/>
+
+Supported uses include:
+
+* Writing of messages to syslog, both local (using the PMIX\_LOG\_LOCAL\_SYSLOG attribute) or on the syslog of some central server designated for this purpose. The PMIx client will send the logging request to its local server. If the PMIX\_LOG\_LOCAL\_SYSLOG attribute is included in the request, then the PMIx server will immediately output the message to the local syslog. If not, or if the PMIX\_LOG\_GLOBAL\_SYSLOG attribute is specified, then the PMIx server will "upcall" the request to its host RM daemon. It is the responsibility of the host RM daemon to identify and transfer the provided data to the appropriate location - upon arrival, the RM can use the PMIx\_Log function to deliver the data to the local syslog on that node, or can write the data to syslog itself. Attributes for setting the desired syslog priority are provided - the default is set to LOG\_INFO indicating reporting of an informational message
```

Comment:
Again I'm thinking towards the standard and how this would transition into standard text.  I would rather see a very concrete presentation of:   PMIx_Log_nb takes the following arguments.  Here is the type of each and what they mean.   Here are the valid acceptable values for each type of argument.   The presentation of these RFC's seem to read like, "Well you can do this, and you can do this, and this will work and behave this way and sometimes you might want to do this".   Maybe its just personal style, but I would much rather see a very concrete list of here are the acceptable arguments and then some text after that giving restrictions on what can and can't be used together.

Reply:
Maybe part of it is that this is an extension to something that already exists.  But in that case I guess maybe my issue is that it is very hard to tell what is new in this RFC and what already exists today.

Reply:
I will break it out as requested